src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,279 @@
+
+        // ToDo: Optimization Idea: Do not send all registered student information but just the number in a DTO
+        var tutorialGroups = tutorialGroupRepository.findAllByCourseIdWithTeachingAssistantAndRegistrations(courseId);
+
+        return ResponseEntity.ok(new ArrayList<>(tutorialGroups));
+    }
+
+    /**
+     * GET /tutorial-groups/:tutorialGroupId : gets the tutorial group with the specified id.
+     *
+     * @param tutorialGroupId the id of the tutorial group to retrieve
+     * @return ResponseEntity with status 200 (OK) and with body the tutorial group
+     */
+    @GetMapping("/tutorial-groups/{tutorialGroupId}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> getOne(@PathVariable Long tutorialGroupId) {
+        log.debug("REST request to get tutorial group: {}", tutorialGroupId);
+        var tutorialGroup = tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroup.getCourse(), null);
+
+        return ResponseEntity.ok().body(tutorialGroup);
+    }
+
+    /**
+     * POST /courses/:courseId/tutorial-groups : creates a new tutorial group.
+     *
+     * @param courseId      the id of the course to which the tutorial group should be added
+     * @param tutorialGroup the tutorial group that should be created
+     * @return ResponseEntity with status 201 (Created) and in the body the new tutorial group
+     */
+    @PostMapping("/courses/{courseId}/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> create(@PathVariable Long courseId, @RequestBody @Valid TutorialGroup tutorialGroup) throws URISyntaxException {
+        log.debug("REST request to create TutorialGroup: {} in course: {}", tutorialGroup, courseId);
+        if (tutorialGroup.getId() != null) {
+            throw new BadRequestException("A new tutorial group cannot already have an ID");
+        }
+
+        var course = courseRepository.findByIdElseThrow(courseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        tutorialGroup.setCourse(course);
+
+        trimStringFields(tutorialGroup);
+        checkTitleIsUnique(tutorialGroup);
+        TutorialGroup persistedTutorialGroup = tutorialGroupRepository.save(tutorialGroup);
+
+        return ResponseEntity.created(new URI("/api/tutorial-groups/" + persistedTutorialGroup.getId())).body(persistedTutorialGroup);
+    }
+
+    /**
+     * DELETE tutorial-groups/:tutorialGroupId : delete a tutorial group.
+     *
+     * @param tutorialGroupId the id of the tutorial group to delete
+     * @return the ResponseEntity with status 204 (NO_CONTENT)
+     */
+    @DeleteMapping("/tutorial-groups/{tutorialGroupId}")
```

Comment:
Here you might want to add the course as path to make it consistent with other endpoints.

Reply:
Done in c926db4b405b0d47b989056509025ea28d1098c5